code/items/weapons/TTTWeapon.cs

```diff
@@ -157,6 +160,22 @@ public override void Simulate(Client owner)
                 }
             }
 
+            if (Input.Released(InputButton.View) && AmmoClip > 0) //Todo: Move this to be a bindable key (default Z???)
+            {
+                if (IsServer)
+                {
+                    var ammoBox = new BuckshotAmmo()
+                    {
+                        Position = Owner.EyePos + Owner.EyeRot.Forward * AmmoDropPositionOffset,
+                        Rotation = Owner.EyeRot
+                    };
+                    ammoBox.Velocity = Owner.EyeRot.Forward * AmmoDropVelocity;
+                    ammoBox.SetCurrentAmmo(AmmoClip);
+                }
+
+                TakeAmmo(AmmoClip);
```

Comment:
We need separate implementations for each weapon right, we don't drop `Buckshot` for every single `TTTWeapon`. Wrap this up in a nice function and implement it for each weapon.

Reply:
Totally forgot about this when I started adding new ammo, whoops. This is why we review! Will get that adjusted.